modules/material/themes/material/default/selectidp-links.twig

```diff
@@ -0,0 +1,129 @@
+<!DOCTYPE html>
+<html>
+<head>
+  <title>{{ '{selectidp-links:title}'|trans }}</title>
+
+  {% include 'header.twig' %}
+
+  <script>
+    function setSelectedIdp(id) {
+      const idpInput = document.createElement('input');
+
+      idpInput.type = 'hidden';
+      idpInput.name = '{{ returnIDParam|e }}';
```

Comment:
No apology needed. It's still new to me too. I think in later templates I ended up using `|e('js')|raw`. This is because I feel a bit better about the js escape than the `json_encode`. The `|raw` part disables the PHP escape that would happen because SSP enables automatic escaping.

Reply:
BTW, the `|e('js')` doesn't add single quotes, so those are still needed.

Reply:
> ... because SSP enables automatic escaping.

I no longer think automatic escaping is enabled by default because I find no evidence of it now. Can't remember where I saw it before to reproduce it.

Reply:
[This](https://twig.symfony.com/doc/3.x/templates.html#html-escaping) is where it said automatic escaping is enabled by default, so yeah... that is confusing.

Reply:
Ahh, but SSP could be overriding/changing that setting. Gotcha.

I'm inclined to think it may be worth [explicitly turning on automatic escaping](https://twig.symfony.com/doc/3.x/templates.html#working-with-automatic-escaping) in each of our twig templates, just to resolve any ambiguity.